stubs/tensorflow/tensorflow/keras/callbacks.pyi

```diff
@@ -0,0 +1,188 @@
+    def on_batch_end(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_epoch_begin(self, epoch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_epoch_end(self, epoch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_predict_batch_begin(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_predict_batch_end(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_predict_begin(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_predict_end(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_test_batch_begin(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_test_batch_end(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_test_begin(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_test_end(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_train_batch_begin(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_train_batch_end(self, batch: int, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_train_begin(self, logs: Mapping[str, object] | None = None) -> None: ...
+    def on_train_end(self, logs: Mapping[str, object] | None = None) -> None: ...
+
+class History(Callback):
+    history: dict[str, list[Any]]
+
+class ModelCheckpoint(Callback):
+    monitor_op: Any
+    filepath: str
+    _options: tf.train.CheckpointOptions | tf.saved_model.SaveOptions | None
+    def __init__(
+        self,
+        filepath: str,
+        monitor: str = "val_loss",
+        verbose: Literal[0, 1] = 0,
+        save_best_only: bool = False,
+        save_weights_only: bool = False,
+        mode: Literal["auto", "min", "max"] = "auto",
+        save_freq: str | int = "epoch",
+        options: tf.train.CheckpointOptions | tf.saved_model.SaveOptions | None = None,
+        initial_value_threshold: float | None = None,
+    ) -> None: ...
+    def _save_model(self, epoch: int, batch: int | None, logs: _Logs) -> None: ...
+
+class TensorBoard(Callback):
+    _start_batch: int
+    _stop_batch: int
+    _should_write_train_graph: bool
+    write_steps_per_second: bool
+    _batch_start_time: float
+    _train_step: tf.Variable
+    _should_trace: bool
+    _is_tracing: bool
+    _global_train_batch: int
+    update_freq: int | Literal["epoch"]
+
+    def __init__(
+        self,
+        log_dir: str = "logs",
+        histogram_freq: int = 0,
+        write_graph: bool = True,
+        write_images: bool = False,
+        write_steps_per_second: bool = False,
+        update_freq: int | Literal["epoch"] = "epoch",
+        profile_batch: int | tuple[int, int] = 0,
+        embeddings_freq: int = 0,
+        embeddings_metadata: dict[str, None] | None = None,
```

Comment:
Missing **kwargs? Although at same time I think all kwargs are always either ignored, deprecated, or an error in the end.

Reply:
kwargs added in c219b9065